src/bin/julialauncher.rs

```diff
@@ -553,6 +1247,48 @@ fn get_override_channel(
     }
 }
 
+/// Determines which channel to use based on the inputs.
+/// This is the core channel selection logic used both in production and tests.
+/// Returns (channel_name, source)
+///
+/// Priority order:
+/// 1. Command line (+channel)
+/// 2. Environment variable (JULIAUP_CHANNEL)
+/// 3. Override (from config file)
+/// 4. Auto-detection (from project manifest)
+/// 5. Default (from config file)
+fn determine_channel(
+    args: &[String],
+    env_channel: Option<String>,
+    override_channel: Option<String>,
+    default_channel: Option<String>,
+    versions_db: &JuliaupVersionDB,
+) -> Result<(String, JuliaupChannelSource)> {
+    // Parse command line for +channel
+    let mut channel_from_cmd_line: Option<String> = None;
+    if args.len() > 1 {
+        let first_arg = &args[1];
+        if let Some(stripped) = first_arg.strip_prefix('+') {
+            channel_from_cmd_line = Some(stripped.to_string());
+        }
+    }
+
+    // Priority order
+    if let Some(channel) = channel_from_cmd_line {
+        Ok((channel, JuliaupChannelSource::CmdLine))
+    } else if let Some(channel) = env_channel {
+        Ok((channel, JuliaupChannelSource::EnvVar))
+    } else if let Some(channel) = override_channel {
+        Ok((channel, JuliaupChannelSource::Override))
+    } else if let Some(channel) = get_auto_channel(args, versions_db) {
+        Ok((channel, JuliaupChannelSource::Auto))
+    } else if let Some(channel) = default_channel {
+        Ok((channel, JuliaupChannelSource::Default))
+    } else {
+        Err(anyhow!("Failed to determine juliaup channel"))
+    }
+}
```

Comment:
This seems like an unrelated refactoring?

Reply:
No, because it makes this part of the code unit testable and we want to test that we get the correct channel in many different cases with the version detection.

Reply:
Not really, it is not needed in a separate PR?

Reply:
Ok not a big issue from me if it doesn't feel right to split it